From 2c7a5a0bdb8e66beb9d9bae0b82bc7e9f6f426c5 Mon Sep 17 00:00:00 2001
From: Johan Hovold <johan+linaro@kernel.org>
Date: Tue, 24 Oct 2023 14:31:06 +0200
Subject: [PATCH 033/173] fixup: USB: dwc3: qcom: add missing interrupt-names
 sanity checks

Add the missing sanity checks when parsing the devicetree
'interrupt-names' property.

Signed-off-by: Johan Hovold <johan+linaro@kernel.org>
---
 drivers/usb/dwc3/dwc3-qcom.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/usb/dwc3/dwc3-qcom.c b/drivers/usb/dwc3/dwc3-qcom.c
index d1b1a9216726..175ee6575320 100644
--- a/drivers/usb/dwc3/dwc3-qcom.c
+++ b/drivers/usb/dwc3/dwc3-qcom.c
@@ -659,12 +659,18 @@ static int dwc3_qcom_setup_irq(struct platform_device *pdev)
 	int i;
 
 	irq_count = of_property_count_strings(np, "interrupt-names");
+	if (irq_count < 0)
+		return irq_count;
+
 	irq_names = devm_kzalloc(&pdev->dev, sizeof(*irq_names) * irq_count, GFP_KERNEL);
 	if (!irq_names)
 		return -ENOMEM;
 
 	ret = of_property_read_string_array(np, "interrupt-names",
 						irq_names, irq_count);
+	if (ret < 0)
+		return ret;
+
 	for (i = 0; i < irq_count; i++) {
 		irq_index = dwc3_qcom_get_irq_index(irq_names[i]);
 		if (irq_index == -1) {
-- 
2.43.0

